### ISSUE TITLE
[5.x]: Invalid purchasable ID in CP create order when order set to site with no products

### ISSUE THREAD

**Issue by bartdigitalpulse** [craftcms/commerce#3756]
### Description

I have a setup of 8 sites (in admin/settings/sites) where 4 of them use commerce. So only 4 of them have products enabled for their site in admin/commerce/settings/producttypes. The primary site is a site that uses products and commerce. 

When I start a new order in the CP the default selected site is a non commerce, non product site. It's probably sorted by Id? I would expect this to be primary site. Every time I start a new order I need to change it manually to the primary site.

When that first non commerce site is selected I'm still able to search products in the add line item. When I try to add them then I get the Invalid purchasable ID because the product is not available for that site.

![Image](https://github.com/user-attachments/assets/c75c030e-9e6b-4e94-833b-498820a2688b)

### Steps to reproduce

1. See above

### Expected behavior

Set primary site as default order site.
Don't show products if they are not enabled for the current order site

### Actual behavior

First site by id is the default order site.
You can search and display products even if they are not enabled for that site.

### Craft CMS version

Craft Pro 5.4.10.1

### Craft Commerce version

5.2.2

### PHP version

8.2.6

### Installed plugins and versions

-

**Comment by nfourtythree** [comment]
Hi @bartdigitalpulse 

Thank you for bringing this to our attention.

We have pushed a fix for the issue which will be included in the next release of Commerce.

To get this early, change your `craftcms/commerce` requirement in your project's `composer.json` to:

```json
"require": {
  "craftcms/commerce": "5.x-dev as 5.2.2.1",
  "...": "..."
}
```

Then run `composer update`.

Thanks!

**Comment by nfourtythree** [comment]
This fix was included in the `5.2.3` release of Commerce

Thanks!
